src/BaseFirestoreRepository.ts

```diff
@@ -42,8 +42,8 @@ export default class BaseFirestoreRepository<T extends IEntity>
     }
   }
 
-  findById(id: string): Promise<T> {
-    return this.firestoreColRef
+  async findById(id: string): Promise<T> {
+    return await this.firestoreColRef
```

Comment:
This `await` is unnecessary given that `extractTFromDocSnap` returns a promise. I could allow the `async` for the sake of consistency though.

Reply:
yeah, this just stuck out to me. our code base at work sticks to always `return await` for consistency, ease of quickly groking return types, and stack tracing errors, which is important. I think the stack trace disappearing without the `await` has been fixed in the latest versions of node. I dunno, the performance difference is next to non-existent and those reasons above. I'm happy to remove it, but just wanted to put some reason behind that choice.

Reply:
I know that the performance difference is practically non existent but for consistency I would remove it